source/ShopPanel.cpp

```diff
@@ -115,10 +115,29 @@ ShopPanel::ShopPanel(PlayerInfo &player, bool isOutfitter)
 
 void ShopPanel::Step()
 {
-	// If the player has acquired a second ship for the first time, explain to
-	// them how to reorder the ships in their fleet.
-	if(player.Ships().size() > 1)
-		DoHelp("multiple ships");
+	if(!checkedHelp && GetUI()->IsTop(this) && player.Ships().size() > 1)
+	{
+		if(DoHelp("multiple ships"))
```

Comment:
I should have said in my previous comment that putting the ship count check here means `checkedHelp` isn't being set to "true" until we have at least two ships. Which means the player might get a help pop up upon buying a second ship.
With the previous implementation, the pop ups would only show up when entering the shop. Even if you open the shipyard with one ship and then buy another, you wouldn't get the help message until you left and came back.
If we want that behaviour, we can do something like:
```suggestion
	if(!checkedHelp && GetUI()->IsTop(this) && player.Ships().size() > 1)
	{
		if(player.Ships().size() < 2)
		{
			// Don't show any help messages.
		}
		else if(DoHelp("multiple ships"))
```
The behaviour before this PR is that you will get the help message upon buying a second ship, so maybe this is fine as-is.

Reply:
yes, I'd keep the current behavior